sdk/python/kfp/dsl/_types.py

```diff
@@ -0,0 +1,205 @@
+					"europe-north1","europe-west1","europe-west2",
+					"europe-west3","europe-west4","northamerica-northeast1",
+					"southamerica-east1","us-central1","us-east1",
+					"us-east4","us-west1", "us-west4" ]
+	}'''
+
+def GCPRegion(attr={}):
+	return type('GCPRegion', (MetaGCPRegion, ), attr)
+
+class MetaGCPProjectID(MetaType):
+	'''MetaGCPProjectID: GCP project id'''
+	openapi_schema_validator = '''{
+		"type": "string"
+	}'''
+
+def GCPProjectID(attr={}):
+	return type('GCPProjectID', (MetaGCPProjectID, ), attr)
+
+# General Types
+class MetaLocalPath(MetaType):
+	#TODO: add restriction to path
+	openapi_schema_validator = '''{
+		"type": "string"
+	}'''
+
+def LocalPath(attr={}):
+	return type('LocalPath', (MetaLocalPath, ), attr)
+
+class InconsistentTypeException(Exception):
+	'''InconsistencyTypeException is raised when two types are not consistent'''
+	pass
+
+def _check_valid_dict(payload):
+	'''_check_valid_dict_type checks whether a dict is a correct serialization of a type
+	Args:
+		payload(dict)
+	'''
+	if not isinstance(payload, dict) or len(payload) != 1:
+		return False
+	for type_name in payload:
+		if not isinstance(payload[type_name], dict):
+			return False
+		property_types = (int, str, float, bool)
+		for property_name in payload[type_name]:
+			if not isinstance(property_name, property_types) or not isinstance(payload[type_name][property_name], property_types):
+				return False
+	return True
+
+def _class_to_dict(payload):
+	'''serialize_type serializes the type instance into a json string
+	Args:
+		payload(type): A class that describes a type
+
+	Return:
+		dict
+	'''
+	attrs = set([i for i in dir(payload) if not i.startswith('__')])
+	json_dict = {payload.__name__: {}}
+	for attr in attrs:
+		json_dict[payload.__name__][attr] = getattr(payload, attr)
+	return json_dict
+
+def _str_to_dict(payload):
+	import json
+	json_dict = json.loads(payload)
+	if not _check_valid_dict(json_dict):
+		raise ValueError(payload + ' is not a valid type string')
+	return json_dict
+
+def _check_dict_types(typeA, typeB):
+	'''_check_type_types checks the type consistency.
+	Args:
+  	typeA (dict): A dict that describes a type from the upstream component output
+  	typeB (dict): A dict that describes a type from the downstream component input
+	'''
+	typeA_name,_ = list(typeA.items())[0]
+	typeB_name,_ = list(typeB.items())[0]
+	if typeA_name != typeB_name:
+		return False
+	type_name = typeA_name
+	for type_property in typeA[type_name]:
+		if type_property not in typeB[type_name]:
+			print(type_name + ' has a property ' + str(type_property) + ' that the latter does not.')
+			return False
+		if typeA[type_name][type_property] != typeB[type_name][type_property]:
+			print(type_name + ' has a property ' + str(type_property) + ' with value: ' +
+						str(typeA[type_name][type_property]) + ' and ' +
+						str(typeB[type_name][type_property]))
+			return False
+	return True
+
+def check_types(typeA, typeB):
```

Comment:
Let's name it `check_argument_type_against_input_type(argument_type, input_type)`?
And for values: `check_argument_value_against_input_type(argument_value, input_type)`.

Reply:
check_types checks the type equality while 
check_argument_type_against_input_type indicates one of the scenarios for this function. Maybe there are other places we need to use this function in the future.

Reply:
But `check_types` does not really tell me what types are being checked.
In some cases that becomes more important. Say, if input type is `TFModel: {version: "<1.8"}` and argument type is `TFModel: {version: "1.6"}`, then they're compatible. But if we switch the types, then they're not quite compatible. Same with type inheritance (I know that we've cut it, but still).

Reply:
Make sense. I guess we have some variations for type checks, inconsistency check, inheritance check, etc. Maybe we can revisit the name when we add more type checking functions. This is not exposed to the user so we will not break anything if we want to rename it.

Reply:
I think it's better to rename the parameter to be typeA -> type_to_check/input_type/argument_type, typeB -> expected_type.

Reply:
SG. Differentiating the typeA and typeB by renaming them makes sense to me.

Reply:
Done.